src/js/popup.js

```diff
@@ -194,6 +194,48 @@ function init() {
     chrome.i18n.getMessage("version", chrome.runtime.getManifest().version)
   );
 
+  $('#expand-blocked-resources').on('click', showBlockedResourcesHandler);
+  $('#collapse-blocked-resources').on('click', hideBlockedResourcesHandler);
+
+  $('#expand-firstparty-popup').on('click', showFirstPartyInfoHandler);
+  $('#collapse-firstparty-popup').on('click', hideFirstPartyInfoHandler);
+
+  if (POPUP_DATA.showExpandedTrackingSection) {
+    $('#expand-blocked-resources').hide();
+    $('#collapse-blocked-resources').show();
+    $('#blockedResources').show();
+  } else if (!POPUP_DATA.showExpandedTrackingSection) {
+    $('#expand-blocked-resources').show();
+    $('#collapse-blocked-resources').hide();
+    $('#blockedResources').hide();
+  }
+
+  $('#instructions-firstparty-description').hide();
+  $('#collapse-firstparty-popup').hide();
+
+  // check if any firstparty scripts are run on current tab & show message in popup
+  async function fetchFirstPartiesManifest() {
+    // fetch the manifest
+    const response = await fetch('../manifest.json');
+    const blob = await response.json();
+
+    // remove the 'www.' from current tab for string matching against first parties manifest url schemes
+    let current_tab = POPUP_DATA.tabHost.slice(4);
+
+    // if current tab is in first parties list, show the popup message
+    for (let firstPartyObj of blob.content_scripts) {
+      firstPartyObj.matches.forEach((urlScheme) => {
+        if (urlScheme.includes(current_tab)) {
```

Comment:
This sort of works but could produce false positives.

What if we leave `tabHost` alone and instead `extractHostFromURL()` from "matches" entries?

- If the extracted host starts with a wildcard, see if `tabHost` ends with host (sliced to remove wildcard).
- If the host does not start with a wildcard, we should see if host and `tabHost` are exactly equal.